src/data_availability/node_state.rs

```diff
@@ -66,43 +79,56 @@ impl NodeState {
-                                verified_proof_transaction.hyle_output.index.clone(),
-                            ));
-                            // Keep track of settled txs
-                            settled_blob_tx_hashes.extend(proof_tx_output.settled_blob_tx_hashes);
-                            // Update the contract's updated states
-                            updated_states.extend(proof_tx_output.updated_states);
-                            // Keep track of all verified proof txs
-                            new_verified_proof_txs.push(tx.clone());
-                        }
-                        Err(e) => {
-                            error!("Failed to handle proof transaction: {:?}", e);
-                            failed_txs.push(tx.clone());
-                        }
-                    }
-                }
-                TransactionData::VerifiedRecursiveProof(rec_proof_tx) => {
+                TransactionData::VerifiedProof(rec_proof_tx) => {
+                    let mut rec_metadata_iter = if let Some(meta) = &rec_proof_tx.recursive_metadata
+                    {
+                        meta.iter()
+                    } else {
+                        std::slice::Iter::default()
+                    };
                     rec_proof_tx
-                        .verifies
+                        .proven_blobs
                         .iter()
                         .for_each(|verified_proof_transaction| {
-                            match self.handle_verified_proof_tx(verified_proof_transaction) {
+                            let contract_name = &match self
+                                .unsettled_transactions
+                                .get(&verified_proof_transaction.blob_tx_hash)
+                                .map(|d| {
+                                    d.blobs
+                                        .get(verified_proof_transaction.hyle_output.index.0)
+                                        .map(|b| b.contract_name.clone())
+                                }) {
+                                Some(Some(contract_name)) => contract_name,
+                                _ => {
+                                    error!("BlobTx contract name not found");
+                                    failed_txs.push(tx.clone());
+                                    return;
+                                }
+                            };
+                            if rec_proof_tx.recursive_metadata.is_some() {
+                                let expected_program_id = rec_metadata_iter.next();
+                                let contract = self.contracts.get(contract_name);
+                                info!(
+                                    "Expected program id: {:?}, contract: {:?}",
+                                    expected_program_id, contract
+                                );
+                                if expected_program_id.is_none()
```

Comment:
Jpense que ca va logger une erreur à chaque fin d'itération; meme si y'a pas d'erreur :smile:

Reply:
Non on ne passe ici que dans le cas récursif où en principe c'est pas none